lxd/seccomp/seccomp.go

```diff
@@ -1359,7 +1364,7 @@ func (s *Server) doDeviceSyscall(c Instance, args *MknodArgs, siov *Iovec) int {
 		return int(-C.EPERM)
 	}
 
-	if !hasCapability {
+	if !bool(C.is_whiteout(args.cDev, args.cMode)) && !hasCapability {
```

Comment:
How about swapping the 2 conditions? `hasCapability` being a `bool` should be cheaper to check so a good candidate.